### PR TITLE
Add debug.sh from openshift-sdn repo

### PR DESCRIPTION
**Pull request by rcarrillocruz**

### CODE REVIEW

File: debug-scripts/debug_sdn.sh [hunk 1]
```diff
@@ -0,0 +1,581 @@
...
+	    aos_master_api_service=$name-master-api.service
+	fi
+	if [ -f $systemd_dir/$name-node.service ]; then
+	    aos_node_service=$name-node.service
+	fi
+    done
+done
+
+case "$1" in
+    --node)
```
Comment (zshi-redhat) [inline debug-scripts/debug_sdn.sh:532]: Is this option for one node testing? I didn't see a loop against nodes in do_node call.

File: debug-scripts/debug_sdn.sh [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+
+logdir=$(mktemp --tmpdir -d openshift-sdn-debug-XXXXXXXXX)
+mkdir $logdir/meta
+cp $self $logdir/meta/debug.sh
+mkdir $logdir/master
+mkdir $logdir/nodes
+do_master_and_nodes "$1" |& tee $logdir/log
+
+dumpname=openshift-sdn-debug-$(date --iso-8601).tgz
+(cd $logdir; tar -cf - --transform='s/^\./openshift-sdn-debug/' .) | gzip -c > $dumpname
```
Comment (zshi-redhat) [inline debug-scripts/debug_sdn.sh:579]: I see there are both echo to stdout and echo to a file, how can we say one log should send to which pipe?

File: debug-scripts/debug_sdn.sh [hunk 3]
```diff
@@ -0,0 +1,581 @@
...
+
+    # Log the generic system stuff
+    log_system $lognode
+
+    # Log some node-only information
+    echo_and_eval  brctl show                              &> $lognode/bridges
+    echo_and_eval  docker ps -a                            &> $lognode/docker-ps
+    echo_and_eval  $OVS_OFCTL -O OpenFlow13 dump-flows br0  &> $lognode/flows
+    echo_and_eval  $OVS_OFCTL -O OpenFlow13 show br0        &> $lognode/ovs-show
+    echo_and_eval  tc qdisc show                           &> $lognode/tc-qdisc
```
Comment (zshi-redhat) [inline debug-scripts/debug_sdn.sh:321]: do we want to organize the output files into different dir or wrap the same type of logs to a single file? for example have all the tc output into the same TC file.

File: debug-scripts/debug_sdn.sh [hunk 4]
```diff
@@ -0,0 +1,581 @@
...
+	if [ -f $systemd_dir/$name-master-api.service ]; then
+	    aos_master_api_service=$name-master-api.service
+	fi
+	if [ -f $systemd_dir/$name-node.service ]; then
+	    aos_node_service=$name-node.service
+	fi
+    done
+done
+
+case "$1" in
```
Comment (zshi-redhat) [inline debug-scripts/debug_sdn.sh:531]: Will this parameter be passed down from image cmd directly or there will be another layer of translation between image cmd params and the debug script params?

**Comment by rcarrillocruz** [comment]
@zshi-redhat I'm just importing https://raw.githubusercontent.com/openshift/openshift-sdn/master/hack/debug.sh onto the image, didn't write it.
I suggest we import it and then iterate it afterwards once more scripts are in in terms of reusing functions and what not.

File: debug-scripts/debug_sdn.sh [hunk 5]
```diff
@@ -0,0 +1,581 @@
+#!/bin/bash
+
+if [ "$(docker ps | grep openvswitch)" ]; then
+    OVS_OFCTL="docker exec openvswitch ovs-ofctl"
+    OVS_APPCTL="docker exec openvswitch ovs-appctl"
+else
+    OVS_OFCTL="ovs-ofctl"
+    OVS_APPCTL="ovs-appctl"
```
Comment (danielmellado) [inline debug-scripts/debug_sdn.sh:8]: Although I see room for further improvements, such as enabling vlog/set debug with OVS_APPCTL, I understand that this is just an import and outside of the scope of this patch. Let's get this merged as-is and iterate from there.

**Review by danielmellado — APPROVED** [review]
Thanks ricky! @zshi-redhat, as this is just an initial import from the already-existing script, let's get this in and iterate from there!

**Comment by zshi-redhat** [comment]
> @zshi-redhat I'm just importing https://raw.githubusercontent.com/openshift/openshift-sdn/master/hack/debug.sh onto the image, didn't write it.
> I suggest we import it and then iterate it afterwards once more scripts are in in terms of reusing functions and what not.

sounds good

**Comment by zshi-redhat** [comment]
> Thanks ricky! @zshi-redhat, as this is just an initial import from the already-existing script, let's get this in and iterate from there!

yeah, let's merge it first then iterate.

**Comment by juanluisvaladas** [comment]
/approve
let's see if now it works :)

**Pull request merged**